R/datasheet.R

```diff
@@ -917,8 +1164,13 @@ setMethod("datasheet",
         names(allScns) <- c("ScenarioId", "ProjectId", "ScenarioName", "ParentId", "ParentName")
         allScns <- allScns[allScns$ScenarioId %in% sid,]
         
-        sheet <- sheet[, !(names(sheet) %in% "ScenarioId")]
-        sheet <- merge(allScns, sheet, all.y = TRUE)
+        # KEEP ScenarioId in sheet for multi-scenario joins
+        # Only drop ScenarioId if single-scenario extraction
+        # NOTE: This endpoint is not reached by single-scenario cases
+        if (length(sid) == 1) {
```

Comment:
I don't think we want to set the ScenarioId to NULL here in any case because this statement is only hit if the conditional above that `returnScenarioInfo=TRUE` is met.